stoppable_workers.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import (
+	"context"
+	"sync"
+)
+
+// StoppableWorkers is a collection of goroutines that can be stopped at a later time.
+type StoppableWorkers interface {
+	AddWorkers(...func(context.Context))
+	Stop()
+	Context() context.Context
+}
+
+// stoppableWorkersImpl is the implementation of StoppableWorkers. The linter will complain if you
+// try to make a copy of something that contains a sync.WaitGroup (and returning a value at the end
+// of NewStoppableWorkers() would make a copy of it), so we do everything through the
+// StoppableWorkers interface to avoid making copies (since interfaces do everything by pointer).
```

Comment:
We do want to return a pointer from the constructor, but we can just have `StoppableWorkers` be a struct and return `*StoppableWorkers` instead of returning an interface. Removed the interface definition.